src/main/java/backend/Logic.java

```diff
@@ -157,32 +160,57 @@ public void refresh() {
         for (Map.Entry<Integer, IssueMetadata> entry : metadata.entrySet()) {
             IssueMetadata currentMetadata = entry.getValue();
             Date lastNonSelfUpdate = new Date(0);
+            Date lastSelfUpdate = new Date(0);
+            boolean isUpdatedByOthers = false;
+            boolean isUpdatedBySelf = false;
             for (TurboIssueEvent event : currentMetadata.getEvents()) {
                 if (!event.getActor().getLogin().equalsIgnoreCase(currentUser)
                         && event.getDate().after(lastNonSelfUpdate)) {
                     lastNonSelfUpdate = event.getDate();
+                    isUpdatedByOthers = true;
+                } else if (event.getActor().getLogin().equalsIgnoreCase(currentUser)
+                        && event.getDate().after(lastSelfUpdate)){
+                    lastSelfUpdate = event.getDate();
+                    isUpdatedBySelf = true;
                 }
             }
             for (Comment comment : currentMetadata.getComments()) {
                 if (!comment.getUser().getLogin().equalsIgnoreCase(currentUser)
                         && comment.getCreatedAt().after(lastNonSelfUpdate)) {
                     lastNonSelfUpdate = comment.getCreatedAt();
+                    isUpdatedByOthers = true;
+                } else if (comment.getUser().getLogin().equalsIgnoreCase(currentUser)
+                        && comment.getCreatedAt().after(lastSelfUpdate)){
+                    lastSelfUpdate = comment.getCreatedAt();
+                    isUpdatedBySelf = true;
                 }
             }
 
             entry.setValue(new IssueMetadata(currentMetadata,
                     LocalDateTime.ofInstant(lastNonSelfUpdate.toInstant(), ZoneId.systemDefault()),
-                    calculateNonSelfCommentCount(currentMetadata.getComments(), currentUser)
+                    LocalDateTime.ofInstant(lastSelfUpdate.toInstant(), ZoneId.systemDefault()),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, IssueType.OTHER_UPDATED),
+                    calculateCommentCount(currentMetadata.getComments(), currentUser, IssueType.SELF_UPDATED),
```

Comment:
All this logic should be in the `IssueMetadata` constructor (or some factory method), and broken down further into different methods. This method is doing too much, plus `IssueMetadata` should handle its own construction.